### PR TITLE
feat: deno completions command

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: cli/flags.rs [hunk 1]
```diff
@@ -590,6 +602,14 @@ pub fn flags_from_vec(
       argv.extend(vec![source_file.to_string(), out_file.to_string()]);
       DenoSubcommand::Bundle
     }
+    ("completions", Some(_completions_match)) => {
+      create_cli_app().gen_completions_to(
+        "deno",
+        Shell::Bash,
+        &mut std::io::stdout(),
+      );
+      std::process::exit(1);
```
Comment (ry) [inline cli/flags.rs]: Why exit(1) ?

Comment (bartlomieju) [reply cli/flags.rs]: No reason, I was prototyping late at night and just copy pasted from other place... I'll ping you when it's ready for review

**Comment by bartlomieju** [comment]
@ry this is now finished

File: cli/flags.rs [hunk 2]
```diff
@@ -300,7 +307,7 @@ ability to spawn subprocesses.
           // this is a fake subcommand - it's used in conjunction with
           // AppSettings:AllowExternalSubcommand to treat it as an
           // entry point script
-          SubCommand::with_name("<script>").about("Script to run"),
+          SubCommand::with_name("[SCRIPT]").about("Script to run"),
```
Comment (bartlomieju) [inline cli/flags.rs]: `bash` doesn't like `<` and `>` in completions, so changed that

File: cli/flags.rs [hunk 3]
```diff
@@ -590,6 +608,15 @@ pub fn flags_from_vec(
...
       DenoSubcommand::Bundle
     }
+    ("completions", Some(completions_match)) => {
+      let shell: &str = completions_match.value_of("shell").unwrap();
+      create_cli_app().gen_completions_to(
+        "deno",
+        Shell::from_str(shell).unwrap(),
+        &mut std::io::stdout(),
+      );
+      std::process::exit(0);
```
Comment (ry) [inline cli/flags.rs]: Is it necessary to exit forcefully here?

File: cli/flags.rs [hunk 4]
```diff
@@ -376,13 +383,24 @@ To change installation directory use -d/--dir flag
           // this is a fake subcommand - it's used in conjunction with
           // AppSettings:AllowExternalSubcommand to treat it as an
           // entry point script
-          SubCommand::with_name("<script>").about("Script URL"),
+          SubCommand::with_name("[SCRIPT]").about("Script URL"),
         ),
     ).subcommand(
+      SubCommand::with_name("completions")
+        .settings(&[
+          AppSettings::DisableHelpSubcommand,
```
Comment (ry) [inline cli/flags.rs]: It would be good to have some help text

File: cli/flags.rs (hunk 3 above)
Comment (bartlomieju) [reply cli/flags.rs]: I can add another `DenoSubcommand` and exit in `main.rs`

File: cli/flags.rs (hunk 4 above)
Comment (ry) [reply cli/flags.rs]: also an entry in the manual

Comment (bartlomieju) [reply cli/flags.rs]: There is help, it's just disabling `deno completions help` command

**Comment by ry** [comment]
Can you apply these adjustments:

```diff
commit a7a4b5672359d1349bd66d45bdb1fc40d0eddb73
Author: Ryan Dahl <ry@tinyclouds.org>
Date:   Tue Jun 25 17:56:49 2019 -0400

    adjust

diff --git a/cli/flags.rs b/cli/flags.rs
index e7fb83ca..f7444f8d 100644
--- a/cli/flags.rs
+++ b/cli/flags.rs
@@ -392,12 +392,12 @@ To change installation directory use -d/--dir flag
           AppSettings::DisableVersion,
         ]).about("Generate shell completions")
         .long_about(
-"Generate and output completion script for shell of choice to standard output.
+"Output shell completion script to standard output.
 
-  deno completions bash > /usr/local/etc/bash_completion.d/deno.bash
+Example:
 
-  source /usr/local/etc/bash_completion.d/deno.bash
-        ")
+  deno completions bash > /usr/local/etc/bash_completion.d/deno.bash
+  source /usr/local/etc/bash_completion.d/deno.bash")
         .arg(
           Arg::with_name("shell")
           .possible_values(&Shell::variants())
diff --git a/third_party b/third_party
index c29d758c..62b961e7 160000
--- a/third_party
+++ b/third_party
@@ -1 +1 @@
-Subproject commit c29d758cd2ca6fd275f141239d59560157ac4535
+Subproject commit 62b961e7d4c40114ffafd728875689918c372e47
diff --git a/website/manual.md b/website/manual.md
index 005a7a57..58f13f42 100644
--- a/website/manual.md
+++ b/website/manual.md
@@ -650,11 +650,11 @@ boolean constant `Deno.noColor`.
 
 ### Shell completion
 
-You can generate completion script for your shell using
-`deno completions <shell>` command. Command outputs to standard output so you
-should redirect it to an appropriate file.
+You can generate completion script for your shell using the
+`deno completions <shell>` command. The command outputs to stdout so you should
+redirect it to an appropriate file.
 
-Currently supported shells are:
+The supported shells are:
 
 - zsh
 - bash
@@ -664,9 +664,9 @@ Currently supported shells are:
 
 Example:
 
-```
-$ deno completions bash > /usr/local/etc/bash_completion.d/deno.bash
-$ source /usr/local/etc/bash_completion.d/deno.bash
+```shellsession
+deno completions bash > /usr/local/etc/bash_completion.d/deno.bash
+source /usr/local/etc/bash_completion.d/deno.bash
 ```
 
 ### V8 flags

```

**Comment by bartlomieju** [comment]
@ry applied

**Pull request merged**